File: SDKs/amazonaws.com/codecommit/2015-04-13/typescript/src/sdk/models/shared/listapprovalruletemplatesoutput.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class ListApprovalRuleTemplatesOutput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=approvalRuleTemplateNames" })
  approvalRuleTemplateNames?: string[];

  @SpeakeasyMetadata({ data: "json, name=nextToken" })
  nextToken?: string;
}

File: SDKs/amazonaws.com/codecommit/2015-04-13/python/src/sdk/models/shared/listapprovalruletemplatesoutput.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class ListApprovalRuleTemplatesOutput:
    approval_rule_template_names: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('approvalRuleTemplateNames') }})
    next_token: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nextToken') }})
    
